src/headers/tomcrypt_pk.h

```diff
@@ -404,6 +404,11 @@ typedef struct {
 } dsa_key;
 
 int dsa_make_key(prng_state *prng, int wprng, int group_size, int modulus_size, dsa_key *key);
+
+int dsa_make_key_ex(prng_state *prng, int wprng, int group_size, int modulus_size, dsa_key *key, char* p_hex, char* q_hex, char* g_hex);
```

Comment:
`dsa_make_params` is basically the same what you do by e.g. `openssl dsaparam 1024 -C`

Looking at my `dsa_make_params` code I am not sure whether the calling interface/API is good (perhaps worth a separate issue "Polish dsa_* API" or so)